packages/fuels-core/src/code_gen/custom_types_gen.rs

```diff
@@ -282,10 +284,10 @@ pub fn expand_custom_enum(enum_name: &str, prop: &Property) -> Result<TokenStrea
         }
 
         impl Parameterize for #enum_ident {
-            fn param_types() -> Vec<ParamType> {
+            fn param_type() -> ParamType {
                 let mut types = Vec::new();
                 #( #param_types; )*
-                types
+                ParamType::Enum(EnumVariants::new(types).unwrap())
```

Comment:
Should we have an `.expect()` with a hint instead of `.unwrap()` here? Feels like this would be a pain to debug if it ever breaks 😅

Reply:
Revisiting this made me see an edge case where we could have a panic in runtime. Kudos! Fixed it in the latest commit.

Also added the message.